FoodPin/FoodPin/Controller/NewRestaurantController.swift

```diff
@@ -117,11 +118,18 @@ class NewRestaurantController: UITableViewController {
             allFieldRequired.addAction(UIAlertAction(title: "Ok", style: .cancel, handler: nil))
             present(allFieldRequired, animated: true, completion: nil)
         } else {
-            print("Name:  \(nameTextField.text ?? "")")
-            print("Type:  \(typeTextField.text ?? "")")
-            print("Location:  \(addressTextField.text ?? "")")
-            print("Phone:  \(phoneTextField.text ?? "")")
-            print("Description:  \(descriptionTextView.text ?? "")")
+            if let appDelegate = (UIApplication.shared.delegate as? AppDelegate) {
+                restaurant = Restaurant(context: appDelegate.persistentContainer.viewContext)
+                restaurant.configure(name: nameTextField.text!, type: typeTextField.text!,
+                                     description: descriptionTextView.text, location: addressTextField.text!,
+                                     phone: phoneTextField.text!, isFavorite: false)
```

Comment:
you should try to not use force unwrapping, you could send an optional instead and populate only if you have the values, or make a validation check first if the values are mandatory